cmake/BLTMacros.cmake

```diff
@@ -270,16 +270,25 @@ macro(blt_add_target_link_flags)
             endif()
         else()
             # In CMake <= 3.12, there is no target_link_flags or target_link_options command
-            get_target_property(_link_flags ${arg_TO} LINK_FLAGS)
-            if(NOT _link_flags)
-                set(_link_flags "")
+            get_target_property(target_type ${arg_TO} TYPE)
+            if (${target_type} STREQUAL "INTERFACE_LIBRARY")
+                # If it's an interface library, we add the flag via link_libraries
+                target_link_libraries(${arg_TO} INTERFACE ${_flags})
```

Comment:
This brings in all INTERFACE properties not just link flags.  I worry about this overreaching and grabbing more than intended.  Though I can't think of anything off hand a case where this would be a problem.  What was the cause of this change?

Reply:
This conditional was added specifically for the `INTERFACE` libraries created by `blt_import_library` - per https://gitlab.kitware.com/cmake/cmake/-/issues/18606, this is how link flags can be propagated through interface libraries.

Interface libraries don't get built, so the only allowable/reasonable access specifier for `target_link_libraries` is `INTERFACE`.

Reply:
https://github.com/LLNL/blt/blob/106e47aeac960c4fac5c5876297b0bba722c6e2d/cmake/BLTMacros.cmake#L481

Header-only libraries also creates an `INTERFACE_LIBRARY`.  I just worry that this macro is only supposed to add link flags and is now adding all `INTERFACE` properties if _flags ends  up having a CMake target in it.

Reply:
To my knowledge, this is the only way to add link options to an interface library pre-3.13 (I think use of this macro before this change would have failed for interface targets with the error message described in the CMake issue I linked).

I'm trying to think of a situation in which a link flag could also be a target, but link flags typically start with `-` or `/` which can't be in the name of a CMake target.  We could always tokenize the flags and look for existing target names, though.

Reply:
no its fine. lets not be crazy of a random thought i had.